infrastructure/main.tf

```diff
@@ -57,11 +63,50 @@ module "custom-network" {
   #firewall_rules = var.firewall_rules
 }
 
+// TODO: Migrate to terraform-google-modules/sql-db/google//modules/private_service_access
+// once https://github.com/terraform-google-modules/terraform-google-sql-db/issues/585 is resolved.
+// We define a VPC peering subnet that will be peered with the
+// Cloud SQL instance network. The Cloud SQL instance will
+// have a private IP within the provided range.
+// https://cloud.google.com/vpc/docs/configure-private-services-access
+
+resource "google_compute_global_address" "google-managed-services-range" {
+  count         = var.create_network ? 1 : 0
+  project       = var.project_id
+  name          = "google-managed-services-${var.network_name}"
+  purpose       = "VPC_PEERING"
+  address_type  = "INTERNAL"
+  prefix_length = 16
+  network       = local.network_self_link
+}
+
+# Creates the peering with the producer network.
+resource "google_service_networking_connection" "private_service_access" {
+  count                   = var.create_network ? 1 : 0
+  network                 = local.network_self_link
+  service                 = "servicenetworking.googleapis.com"
+  reserved_peering_ranges = [google_compute_global_address.google-managed-services-range[0].name]
+  # This will enable a successful terraform destroy when destroying CloudSQL instances
+  deletion_policy = "ABANDON"
+}
+
+// module "private-service-access" {
```

Comment:
we discussed just using this instead

Reply:
I kept this in as VPC deletion seems to work successfully. I was able to run `terraform apply`, go through the demo steps, and then run `terraform destroy` successfully.

Reply:
ah, due to the ABANDON? makes sense!

Reply:
maybe leave a comment explaining why we need the ABANDON and why we didn't use the PSC module, so it doesn't get accidentally deleted in the future

Reply:
Updated the comment